fe/fe-core/src/main/java/com/starrocks/sql/parser/SqlParser.java

```diff
@@ -16,6 +17,55 @@
 import java.util.List;
 
 public class SqlParser {
+
+    public static List<StatementBase> parse(String originSql, SessionVariable sessionVariable) {
+        List<String> splitSql = splitSQL(originSql);
+        List<StatementBase> statements = Lists.newArrayList();
+        for (int idx = 0; idx < splitSql.size(); ++idx) {
+            String sql = splitSql.get(idx);
+            StatementBase statement = parseSingleSql(sql, sessionVariable);
+            statement.setOrigStmt(new OriginStatement(sql, idx));
+            statements.add(statement);
+        }
+        return statements;
+    }
+
+    public static StatementBase parseSingleSql(String sql, SessionVariable sessionVariable) {
+        StarRocksLexer lexer = new StarRocksLexer(new CaseInsensitiveStream(CharStreams.fromString(sql)));
+        CommonTokenStream tokenStream = new CommonTokenStream(lexer);
+        StarRocksParser parser = new StarRocksParser(tokenStream);
+        setParserProperty(parser, sessionVariable);
+        StatementBase statement;
+        try {
+            StarRocksParser.SqlStatementsContext sqlStatements = parser.sqlStatements();
+            statement = (StatementBase) new AstBuilder(sessionVariable.getSqlMode())
+                    .visitSingleStatement(sqlStatements.singleStatement(0));
+            return statement;
+        } catch (OperationNotAllowedException e) {
+            // sql forbidden to execute, so no need to parse again by the old parser.
+            throw e;
+        } catch (ParsingException parsingException) {
+            try {
+                statement = parseWithOldParser(sql, sessionVariable.getSqlMode(), 0);
+            } catch (Exception e) {
+                // both new and old parser failed. We return new parser error info to client.
+                throw parsingException;
+            }
+            if (StatementPlanner.supportedByNewPlanner(statement)) {
+                throw parsingException;
+            }
+            return statement;
+        }
+    }
+
+    public static void setParserProperty(StarRocksParser parser, SessionVariable sessionVariable) {
+        StarRocksParser.sqlMode = sessionVariable.getSqlMode();
+        parser.removeErrorListeners();
+        parser.addErrorListener(new ErrorHandler());
+        parser.removeParseListeners();
+        parser.addParseListener(new TokenNumberListener(sessionVariable.getParseTokensLimit()));
+    }
+
     public static List<StatementBase> parse(String originSql, long sqlMode) {
```

Comment:
please delete this

Reply:
Replaced it in  most scenes. Some scenes is not easy to replace, so I have changed its implemention by invoke new parser method in its body.